graphics/src/widget/canvas/frame.rs

```diff
@@ -262,6 +268,30 @@ impl Frame {
         self.transforms.current.is_identity = false;
     }
 
+    /// Produces a [`Geometry`] representing everything drawn on the [`Frame`], clears the current
+    /// primitives.
+    pub fn geometry(&mut self) -> Geometry {
+        if !self.buffers.indices.is_empty() {
+            self.primitives.push(Primitive::Mesh2D {
+                buffers: triangle::Mesh2D {
+                    vertices: self.buffers.vertices.clone(),
+                    indices: self.buffers.indices.clone(),
+                },
+                size: self.size,
+            });
```

Comment:
Cloning all the vertices and indices just to avoid reallocating the tessellators feels counter-productive.